hooks/checkout

```diff
@@ -40,6 +40,36 @@ check_set() {
   fi
 }
 
+
+# Works like GNU timeout, but implementing it here lets it run on macOS agents.
+# Mostly copied from here: https://stackoverflow.com/a/24413646
+timeout() {
+  local -r duration=$1
+  shift
+  local -r cmd=( "$@" )
+
+  # Run in a subshell to avoid job control messages
+  (
```

Comment:
I think you need `set +e` before this, and `set -e` inside it and after it so it doesn't bubble up on failure.

Reply:
I didn't test it with `set -euo pipefail` until you pointed this out, so thanks. The behaviour is counter-intuitive to me though, because the parenthesised part _doesn't_ cause the script to exit early when the command which `timeout` runs exits unsuccessfully.